### PR TITLE
fix: adopt new version of curl-cffi

### PR DESCRIPTION
**Pull request by vdusek**
### Description

- Version 1.7.2 of `curl-cffi` introduces breaking changes.
- This update includes adopting the new version, adding type aliases for `Request` fields, and incorporating other minor changes from PR #542.

### Issues

- N/A (ad-hoc fix)

### Testing

- The current set of unit tests should cover the changes.

### Checklist

- [x] CI passed

### CODE REVIEW

File: src/crawlee/http_clients/curl_impersonate.py [hunk 1]
```diff
@@ -194,7 +195,7 @@ def _get_client(self, proxy_url: str | None) -> AsyncSession:
             # are set as default options.
             kwargs: dict[str, Any] = {
                 'proxy': proxy_url,
-                'impersonate': BrowserType.chrome,
+                'impersonate': BrowserType.chrome124,
```
Comment (janbuchar) [inline src/crawlee/http_clients/curl_impersonate.py:198]: Isn't there something like a "newest chrome" option?

**Review by janbuchar — APPROVED** [review]
LGTM, one thing to clarify

File: src/crawlee/http_clients/curl_impersonate.py (hunk 1 above)
Comment (vdusek) [reply src/crawlee/http_clients/curl_impersonate.py]: Unfortunately, no. The `BrowserType.chrome` was the most recent option, but it has been removed. However, I discovered that there is a `DEFAULT_CHROME` constant that represents the latest version, so we can use that instead.

**Comment by B4nan** [comment]
lol, so even in python world maintainers don't care about semver :D

**Comment by janbuchar** [comment]
> lol, so even in python world maintainers don't care about semver :D

I don't think phenomenon is language-specific :smile:

**Pull request merged**